chidioguejiofor/airtech-api
# #165484883: User confirm email

#### What does this PR achieve?

- sets up SendGrid for sending emails
- modifies signup route to send the user a confirmation email
- adds a route to confirm their email
- adds an endpoint to resend the email
- cleans up several parts of the codebase
- add tests for the new endpoint using Unittest mocks
- ensures that all tests added are passing


#### Any background context?
- The MailGun setup that was used initially has now been discarded. All emails would be sent using SendGrid from this PR

#### New environmental Variables?
- SENDGRID_API_KEY

#### Manual Testing Steps
- Migrate your data to your database via `python manage.py makemigrations && python manage.py migrate`
- Run Redis and Celery( See [README](https://github.com/chidioguejiofor/airtech-api) for instructions)
- Sign up a user
- You should get a  comfirmation email
